packages/block-library/src/post-date/edit.js

```diff
@@ -62,12 +62,13 @@ export default function PostDateEdit( {
 		'site',
 		'time_format'
 	);
-	const [ date, setDate ] = useEntityProp(
+	const [ postTypeDate, setDate ] = useEntityProp(
 		'postType',
 		postTypeSlug,
 		displayType,
 		postId
 	);
+	const date = postTypeDate || new Date();
```

Comment:
Thanks for trying this out.

For templates it might cause a bit of confusion, where the date doesn't correspond with the actual post date:

https://github.com/WordPress/gutenberg/assets/6458278/20ee5b67-6011-4a37-91d2-da4ebdd6f4f9

Also, the field can be toggled to be the modified date.

I wonder if it's possible to grab the actual post and modified date from the entity record?